docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_expiry_in_30_days.mdx

```diff
@@ -24,6 +24,262 @@ CBP
+
+4. To find out if a reserved node lease is expiring in the next 30 days, you need to write a script that calculates the difference between the current date and the start time plus the duration. If the difference is less than or equal to 30 days, then the lease is expiring in the next 30 days. Here is a simple Python script that does this:
+
+   ```python
+   import json
+   import subprocess
+   from datetime import datetime, timedelta
+
+   # Run the AWS CLI command and get the output
+   output = subprocess.check_output(["aws", "redshift", "describe-reserved-nodes"])
+   data = json.loads(output)
+
+   # Get the current date
+   now = datetime.now()
+
+   # Check each reserved node
+   for node in data['ReservedNodes']:
+       # Get the start time and duration
+       start_time = datetime.strptime(node['StartTime'], '%Y-%m-%dT%H:%M:%S.%fZ')
+       duration = timedelta(seconds=node['Duration'])
+
+       # Calculate the expiration date
+       expiration_date = start_time + duration
+
+       # Check if the lease is expiring in the next 30 days
+       if now <= expiration_date <= now + timedelta(days=30):
+           print(f"Reserved node {node['ReservedNodeId']} is expiring in the next 30 days.")
+   ```
+   
+Please note that this script assumes that the AWS CLI is configured with access to the necessary resources. If it's not, you may need to provide the necessary credentials or configure the AWS CLI accordingly.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+2. Import the necessary libraries and establish a session with AWS using your access keys. 
+
+```python
+import boto3
+from datetime import datetime, timedelta
+
+# Create a session using your AWS credentials
+aws_session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2' # or your preferred region
+)
+```
+
+3. Use the `describe_reserved_nodes` method of the Redshift client to get a list of all reserved nodes. Then, iterate over the list and check the `EndTime` of each reserved node. If the `EndTime` is within the next 30 days, print the details of the node.
+
+```python
+# Create a Redshift client
+redshift = aws_session.client('redshift')
+
+# Get a list of all reserved nodes
+response = redshift.describe_reserved_nodes()
+
+# Get the current date and time
+now = datetime.now()
+
+# Iterate over the reserved nodes
+for node in response['ReservedNodes']:
+    # Get the end time of the reserved node
+    end_time = node['EndTime']
+    
+    # Check if the end time is within the next 30 days
+    if now <= end_time <= now + timedelta(days=30):
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>